extension/js/content_scripts/webmail/setup-webmail-content-script.ts

```diff
@@ -286,17 +286,24 @@ export const contentScriptSetupIfVacant = async (webmailSpecific: WebmailSpecifi
     }
   };
 
-  const startPullingKeysFromEkm = async (acctEmail: string, clientConfiguration: ClientConfiguration, factory: XssSafeFactory, ppEvent: { entered?: boolean }) => {
-    if (clientConfiguration.usesKeyManager()) {
-      const idToken = await InMemoryStore.get(acctEmail, InMemoryStoreKeys.ID_TOKEN);
-      if (idToken) {
-        const keyManager = new KeyManager(clientConfiguration.getKeyManagerUrlForPrivateKeys()!);
-        Catch.setHandledTimeout(async () => {
-          const { privateKeys } = await keyManager.getPrivateKeys(idToken);
-          await processKeysFromEkm(acctEmail, privateKeys.map(entry => entry.decryptedPrivateKey), clientConfiguration, factory, idToken, ppEvent);
-        }, 0);
+  const startPullingKeysFromEkm = async (acctEmail: string, clientConfiguration: ClientConfiguration, factory: XssSafeFactory, ppEvent: { entered?: boolean }): Promise<void> => {
```

Comment:
So this function is no longer `startPulling...` due to the latest modifications, but actually `pull...`?
I can't see a good reason why we should await for the completion of  `getPrivateKeys` and `processKeysFromEkm` before starting the web client. We can simply pass an additional function to call after line 297 in a `try {} catch {} finally {}` block

Reply:
Updated